controllers/vspheremachine_controller.go

```diff
@@ -345,14 +348,22 @@ func (r *machineReconciler) patchMachineLabelsWithHostInfo(ctx context.MachineCo
 		return err
 	}
 
+	info := util.SanitizeHostInfoLabel(hostInfo)
+	errs := validation.IsValidLabelValue(info)
+	if len(errs) > 0 {
+		err := errors.Errorf("%s: %s", hostInfoErrStr, strings.Join(errs, ","))
+		r.Logger.Error(err, hostInfoErrStr, "info", hostInfo)
+		return err
+	}
+
 	machine := ctx.GetMachine()
 	patchHelper, err := patch.NewHelper(machine, r.Client)
 	if err != nil {
 		return err
 	}
 
 	labels := machine.GetLabels()
-	labels[constants.ESXiHostInfoLabel] = hostInfo
+	labels[constants.ESXiHostInfoLabel] = info
```

Comment:
Will truncate the host info label affect anti-affinity logic? For example, if after sanitize, the hostInfoLabel are the same for two esxi hosts, will it affect our anti-affinity logic?
Another question, if DRS kick in, the VMs is vMotioned to another host, will this label get updated accordingly?

Reply:
Nope, the anti affinity logic is independent of the label. The logic is to get the host info from the VM and add it as a label to the Machine object. Eventually, on the next reconcile, the label on the Machine object will be updated to display the latest info.